flycheck.el

```diff
@@ -6082,6 +6082,18 @@ objects)."
 
 
 ;;; Convenience definition of command-syntax checkers
+(eval-and-compile
+  ;; We need the byte-compiler to be aware of this function when compiling
+  ;; `flycheck-define-checker'.
+  (defun flycheck-maybe-quote (arg)
+    "Quote ARG if it is a symbol, return ARG otherwise."
+    (let ((a arg))
+      (if (and (symbolp a) (not (null a)))
```

Comment:
Isn't `not null a` the same as `a`?
Also, why do you let-bind `a`?

Reply:
Ah, I started wrting it as a macro, then found it was unnecessary.